uxarray/io/_mpas.py

```diff
@@ -189,112 +243,348 @@ def _dual_to_ugrid(in_ds, out_ds):
         })
 
     out_ds['Mesh2_face_y'] = xr.DataArray(
-        latVertex,
+        face_lat,
         dims=["nMesh2_face"],
         attrs={
             "standard_name": "latitude",
             "long_name": "latitude of center nodes",
             "units": "degrees_north",
         })
 
-    # vertex indices that surround each dual-mesh cell
-    cellsOnVertex = np.array(in_ds['cellsOnVertex'].values, dtype=INT_DTYPE)
 
-    # replace missing/zero values with fill values
-    _replace_zeros(cellsOnVertex)
+def _parse_face_xyz_coords(in_ds, out_ds, mesh_type):
+    """Parses cartesian face center coordinates for either the Primal or Dual
+    Mesh."""
+    if mesh_type == "primal":
+        face_x = in_ds['xCell'].values
+        face_y = in_ds['yCell'].values
+        face_z = in_ds['zCell'].values
+    else:
+        # centers of dual-mesh cells (in degrees)
+        face_x = in_ds['xVertex'].values
+        face_y = in_ds['yVertex'].values
+        face_z = in_ds['zVertex'].values
 
-    # convert to zero-indexed
-    _to_zero_index(cellsOnVertex)
+    out_ds["Mesh2_face_cart_x"] = xr.DataArray(
+        data=face_x,
+        dims=["nMesh2_face"],
+        attrs={
+            "standard_name": "x",
+            "long_name": "cartesian edge x",
+            "units": "m",
+        })
 
-    out_ds["Mesh2_face_nodes"] = xr.DataArray(
-        data=cellsOnVertex,
+    out_ds["Mesh2_face_cart_y"] = xr.DataArray(
+        data=face_y,
+        dims=["nMesh2_face"],
+        attrs={
+            "standard_name": "y",
+            "long_name": "cartesian edge y",
+            "units": "m",
+        })
+
+    out_ds["Mesh2_face_cart_z"] = xr.DataArray(
+        data=face_z,
+        dims=["nMesh2_face"],
+        attrs={
+            "standard_name": "z",
+            "long_name": "cartesian edge z",
+            "units": "m",
+        })
+
+
+def _parse_edge_latlon_coords(in_ds, out_ds, mesh_type):
+    """Parses latlon edge node coordinates for either the Primal or Dual
+    Mesh."""
+    if mesh_type == "primal":
+        edge_lon = np.rad2deg(in_ds['lonEdge'].values)
+        edge_lat = np.rad2deg(in_ds['latEdge'].values)
+    else:
+        edge_lon = np.rad2deg(in_ds['lonEdge'].values)
+        edge_lat = np.rad2deg(in_ds['latEdge'].values)
```

Comment:
These are identical, so:
```suggestion
    edge_lon = np.rad2deg(in_ds['lonEdge'].values)
    edge_lat = np.rad2deg(in_ds['latEdge'].values)
```